src/test/java/hudson/plugins/tasks/workflow/WorkflowCompatibilityTest.java

```diff
@@ -0,0 +1,82 @@
+package hudson.plugins.tasks.workflow;
+
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import hudson.FilePath;
+
+import hudson.model.Result;
+
+import hudson.plugins.tasks.TasksResultAction;
+
+import static org.junit.Assert.*;
+
+/**
+ * Test workflow compatibility.
+ */
+public class WorkflowCompatibilityTest {
+
+    @ClassRule
+    public static JenkinsRule j = new JenkinsRule();
+
+    private static final String TASKS_FILE = "/hudson/plugins/tasks/parser/file-with-tasks.txt";
+
+    /**
+     * Run a workflow job using {@link TasksPublisher} and check for success.
+     */
+    @Test
+    public void tasksPublisherWorkflowStep() throws Exception {
+        WorkflowJob job = j.jenkins.createProject(WorkflowJob.class, "wf");
+        FilePath workspace = j.jenkins.getWorkspaceFor(job);
+        FilePath report = workspace.child("target").child("tasks.txt");
+        report.copyFrom(WorkflowCompatibilityTest.class.getResourceAsStream(TASKS_FILE));
+        job.setDefinition(new CpsFlowDefinition(
+        "node {" +
+        "  step([$class: 'TasksPublisher', pattern: '**/tasks.txt', high: 'FIXME', normal: 'TODO'])" +
+        "}"));
+        j.assertBuildStatusSuccess(job.scheduleBuild2(0));
+        TasksResultAction result = job.getLastBuild().getAction(TasksResultAction.class);
+        assertTrue(result.getResult().getAnnotations().size() == 2);
+    }
```

Comment:
NIT: @amuniz I don't want to be a PITA, but would you mind adding a comment in the setup e.g. why you are `report.copyFrom(WorkflowCompatibilityTest.class.getResourceAsStream(TASKS_FILE))` and how that helps the test setup and what comes after. No worries if it can't be done easily.